### PR TITLE
Npa

### PR DESCRIPTION
**Pull request by nielsdenissen**
Adds support for NPAs that don't have a sessiontoken.
Also uses specific types for the User object instead of String.

### CODE REVIEW

**Review by arempter — CHANGES_REQUESTED** [review]
I get error while trying to upload file as regular user

Outgoing request stream error
[info] java.lang.IllegalStateException: Substream Source cannot be materialized more than once

Seems that error appears when user is not promoted to system user. Once I run modify --system error is gone

File: src/main/scala/com/ing/wbaa/gargoyle/proxy/api/ProxyService.scala [hunk 1]
```diff
@@ -35,34 +37,26 @@ trait ProxyService extends LazyLogging {
...
-                    logger.error("An error occurred retrieving the User from STS service", exception)
-                    complete(StatusCodes.InternalServerError)
+            onComplete(areCredentialsActive(s3Request.credential)) {
+              case Success(Some(userSTS: User)) =>
+                logger.debug(s"Credentials active for request, user retrieved: $userSTS")
+
+                if (isUserAuthenticated(httpRequest, userSTS.secretKey)) {
+                  logger.debug(s"Request authenticated: $httpRequest")
+
+                  if (isUserAuthorizedForRequest(s3Request, userSTS)) {
```
Comment (arempter) [inline src/main/scala/com/ing/wbaa/gargoyle/proxy/api/ProxyService.scala:47]: I am hitting this with npa user, seq:
Proxy
Signertype used: Some(AWS)
Extracted S3 Request: 
S3Request(AwsRequestCredential(AwsAccessKey(accesskey),None),Some(admin),Some(user),Read)
Credentials active for request, user retrieved:
Request authenticated: HttpRequest
checking ranger with request: RangerAccessRequestImpl=
User (UserName(ceph-admin)) not authorized for request

Sts
isCredentialActive ok for accessKey=accesskey, sessionToken=None

Comment (nielsdenissen) [reply src/main/scala/com/ing/wbaa/gargoyle/proxy/api/ProxyService.scala:47]: Seems like there is no ranger entry that allows `ceph-admin` to do this Read right?

Comment (arempter) [reply src/main/scala/com/ing/wbaa/gargoyle/proxy/api/ProxyService.scala:47]: Problem related to how plugin forms url to talk to RGW (radosgw-4j library related)

**Review by arempter — APPROVED** [review]

**Pull request merged**